source/val/validate.cpp

```diff
@@ -399,6 +398,11 @@ spv_result_t ValidateBinaryUsingContextAndValidationState(
   // Catch undefined forward references before performing further checks.
   if (auto error = ValidateForwardDecls(*vstate)) return error;
 
+  for (size_t i = 0; i < vstate->ordered_instructions().size(); ++i) {
```

Comment:
Please add a comment to explain why this is here so we're not tempted to move it later.

Reply:
Done